addons/pkg/util/cluster_util.go

```diff
@@ -107,37 +107,26 @@ func GetClusterClient(ctx context.Context, currentClusterClient client.Client, s
 	return c, nil
 }
 
-// GetTKRForCluster gets the TKR for cluster
-func GetTKRForCluster(ctx context.Context, c client.Client, cluster *clusterv1beta1.Cluster) (*runtanzuv1alpha1.TanzuKubernetesRelease, error) {
+// GetTKRNameForCluster get the TKR name for the cluster. The tkrLabel argument determines the key of the TKR label
+// only "legacy-style" TKGm clusters have "tanzuKubernetesRelease" label for the TKR, whereas "classy" UTKG
+// clusters have "run.tanzu.vmware.com/tkr" label for the TKR, or not labeled with any TKR-related labels
+func GetTKRNameForCluster(ctx context.Context, c client.Client, cluster *clusterv1beta1.Cluster, tkrLabel string) string {
 	if c == nil || cluster == nil {
-		return nil, nil
-	}
-
-	tkrName := GetTKRNameForCluster(ctx, c, cluster)
-	if tkrName == "" {
-		return nil, nil
-	}
-
-	tkr, err := GetTKRByName(ctx, c, tkrName)
-	if err != nil {
-		return nil, err
+		return ""
 	}
 
-	return tkr, nil
-}
-
-// GetTKRNameForCluster get the TKR name for the cluster
-func GetTKRNameForCluster(ctx context.Context, c client.Client, cluster *clusterv1beta1.Cluster) string {
-	if c == nil || cluster == nil {
-		return ""
+	for k, v := range cluster.Labels {
+		if k == tkrLabel {
+			return v
+		}
 	}
 
-	return cluster.Labels[constants.TKRLabel]
+	return ""
 }
 
-// GetBOMForCluster gets the bom associated with the cluster
+// GetBOMForCluster gets the bom associated with the legacy-style TKGm cluster
 func GetBOMForCluster(ctx context.Context, c client.Client, cluster *clusterv1beta1.Cluster) (*bomtypes.Bom, error) {
-	tkrName := GetTKRNameForCluster(ctx, c, cluster)
+	tkrName := GetTKRNameForCluster(ctx, c, cluster, constants.TKRLabel)
```

Comment:
There's only one usage of this func: from `(r *AddonReconciler) reconcileNormal()`. Again, at the time we're in reconcileNormal we already know the TKR name, so no need to obtain and check for it here.

Reply:
In fact, this function should now become useless. So, it's probably a good idea to deprecate it.

Reply:
Well, TKR versions v1alpha1 still need it?